index.php

```diff
@@ -17,10 +18,66 @@ function grabApi () {
         }
         $i ++;
         curl_setopt($curl, CURLOPT_URL, 'https://pokeapi.co/api/v2/pokemon/' . $i . '/');
-        $pokemon[] = curl_exec($curl);
+        $resp = curl_exec($curl);
+        $temp = json_decode($resp, TRUE);
+        $name = $temp['forms'][0]['name'];
+        $type1 = $temp['types'][0]['type']['name'];
+        $type2 = $temp['types'][1]['type']['name'];
+        $pokemon[] = [$name, $type1, $type2];
     }
     curl_close($curl);
     return $pokemon;
 }
 
-grabApi();
+/*
+ * Allows us to insert all of our data in our DB at once. By putting each into a () we can concatinate each and then insert into the db as one large value.
+ *
+ *$allPokemen - the array of all the pokemon (with the only data we needed).
+ *
+ * $values - stores the concacinated arrays.
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
+ */
+
+function insertintodatabase($allPokemon) {
+    $values = '';
+
+    foreach ($allPokemon as $result){
+        $values .= '(\'' . $result[0] . '\',';
+        $values .= '\'' . $result[1] . '\',';
+       if (isset($result[2])) {
+           $values .= '\'' . $result[2] . '\'), ';
+       } else {
+           $values .= '\'' . 'NULL' . '\'), ';
+       }
+    }
+
+
+    $finalValues = rtrim($values, ', ');
+    return $finalValues;
+}
+
+
+/*
+ * inserts our Pokemon data in the database
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
+ * $db - PDO to our database;
+ */
+
+function database ($finalValues, $db){
+
+    $query = $db->prepare("INSERT INTO `pokemon` (`pokemon_Name`, `pokemon_type`, `pokemon_type2`) VALUES " . $finalValues .";");
```

Comment:
query needs refactoring: pokemon_Name to pokemon_name; pokemon_type2 to pokemon_type_2